frontend/src/metabase/list-view/components/ListView/ColumnValue.tsx

```diff
@@ -1,91 +1,286 @@
+  RowValues,
+} from "metabase-types/api";
 
 import styles from "./ListView.module.css";
-
-// Light background colors for category values
-const CATEGORY_COLORS = [
-  "color-mix(in srgb, var(--mb-color-brand) 8%, var(--mb-color-background))",
-  "color-mix(in srgb, var(--mb-color-success) 8%, var(--mb-color-background))",
-  "color-mix(in srgb, var(--mb-color-warning) 8%, var(--mb-color-background))",
-  "color-mix(in srgb, var(--mb-color-error) 8%, var(--mb-color-background))",
-  "color-mix(in srgb, var(--mb-color-filter) 8%, var(--mb-color-background))",
-  "color-mix(in srgb, var(--mb-color-summarize) 8%, var(--mb-color-background))",
-  "color-mix(in srgb, var(--mb-color-focus) 8%, var(--mb-color-background))",
-  "color-mix(in srgb, var(--mb-color-text-medium) 8%, var(--mb-color-background))",
-];
-
-// Get a consistent color for a category value based on its hash
-const getCategoryColor = (value: any, columnName: string) => {
-  if (value == null || value === "") {
-    return "var(--mb-color-background-light)";
-  }
-
-  const stringValue = String(value);
-
-  // Use a combination of column name and value for more consistent colors
-  const combinedString = `${columnName}:${stringValue}`;
-  const hash = combinedString.split("").reduce((a, b) => {
-    a = (a << 5) - a + b.charCodeAt(0);
-    return a & a;
-  }, 0);
-
-  const colorIndex = Math.abs(hash) % CATEGORY_COLORS.length;
-  return CATEGORY_COLORS[colorIndex];
-};
+import { getCategoryColor } from "./styling";
 
 interface ColumnValueProps {
   column: DatasetColumn;
-  value: ReactNode;
+  settings: ComputedVisualizationSettings;
   rawValue: any;
+  style?: React.CSSProperties;
+  rows: RowValues;
+  cols: DatasetColumn[];
 }
 
-export function ColumnValue({ column, value, rawValue }: ColumnValueProps) {
-  const isBooleanColumn = column.base_type === "type/Boolean";
-  const isCategoryColumn = column.semantic_type === "type/Category";
-  const isScoreColumn = column.semantic_type === "type/Score";
-  const shouldGetCategoryStyling = isCategoryColumn || isScoreColumn;
+const DEFAULT_COLUMN_SETTINGS: ColumnSettings = {};
+export function ColumnValue({
+  column,
+  settings,
+  rawValue,
+  style,
+  rows,
+  cols,
+}: ColumnValueProps) {
+  const columnSettings = useMemo(
+    () => settings.column?.(column),
+    [column, settings],
+  );
+  const value = useMemo(
+    () =>
+      formatValue(rawValue, {
+        ...(columnSettings || DEFAULT_COLUMN_SETTINGS),
+        jsx: true,
+        rich: true,
+      }),
+    [rawValue, columnSettings],
+  );
+  const columnExtent = useMemo(() => {
+    if (isQuantity(column) || isScore(column)) {
+      return getColumnExtent(cols, rows, cols.indexOf(column));
+    }
+    return null;
+  }, [cols, rows, column]);
+
+  // Need to return empty element here to preserve grid column layout in ListViewItem.
+  if (rawValue == null) {
+    return <div />;
+  }
 
-  if (isBooleanColumn) {
+  if (column.base_type === TYPE.Boolean) {
```

Comment:
:x: Please use functions from `frontend/src/metabase-lib/v1/types/utils/isa.js`.

The types should never be compared like this, the comparison should always go through cljs `isa`: https://github.com/metabase/metabase/blob/f15ed1a7341d55b60f0d40a531807d2addda24e6/frontend/src/metabase-lib/v1/types/utils/isa.js#L19-L31